src/main/java/uk/gov/ons/census/fwmtadapter/services/CaseService.java

```diff
@@ -0,0 +1,23 @@
+package uk.gov.ons.census.fwmtadapter.services;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.ons.census.fwmtadapter.model.dto.CaseIdDto;
+
+@Service
+public class CaseService {
+  @Value("${caseapi.host}")
+  private String host;
+
+  @Value("${caseapi.port}")
+  private String port;
+
+  public String getCaseIdFromQid(String questionnaire_id) {
+    String url = "http://" + host + ":" + port + "/cases/qid/" + questionnaire_id;
+    RestTemplate restTemplate = new RestTemplate();
+
+    CaseIdDto caseIdDto = restTemplate.getForObject(url, CaseIdDto.class);
+    return caseIdDto.getCaseId();
```

Comment:
Minor comment, but would we want to handle if a case ID isn't found here?

Reply:
The returned object contains only a case Id and only returns it if successfully found.  However I will add a null check and throw RuntimeException if null.